integration/kotlinx-coroutines-guava/test/ListenableFutureTest.kt

```diff
@@ -205,6 +336,59 @@ class ListenableFutureTest : TestBase() {
         finish(4)
     }
 
+    @Test
+    fun testCancelledDeferredAsListenableFutureAwaitThrowsCancellation() = runTest {
+        val future = Futures.immediateCancelledFuture<Int>()
+        val asDeferred = future.asDeferred()
+        val asDeferredAsFuture = asDeferred.asListenableFuture()
+
+        assertTrue(asDeferredAsFuture.isCancelled)
+        assertFailsWith<CancellationException> {
+            val value: Int = asDeferredAsFuture.await()
+        }
+    }
+
+    @Test
+    fun testCancelledDeferredAsListenableFutureAsDeferredJoinDoesNotThrow() = runTest {
```

Comment:
`join` throws if and only if a caller of the `join` was cancelled

Reply:
Am I understanding right that you mean the test case's name is redundant? I think I see tha. I've picked a new name to make it clearer what I'm trying to test for here.

To paraphrase, my intention is to test that the correct `Deferred` cancellation completion behaviour tunnels through a `Deferred` -> `Future` -> `Deferred` transformation. That's a good guarantee that the two transforms interoperate properly.

Let me know if I've misunderstood your perspective here.